app/src/main/java/com/ivy/wallet/ui/RootActivity.kt

```diff
@@ -179,6 +175,10 @@ class RootActivity : AppCompatActivity(), RootScreen {
                 test = { TestScreen() }
             )
         )
+        // Check if the intent was triggered by a shortcut action, and notify the view model of the shortcut click event
+        if (intent.action == SHORTCUT_ACTION) {
```

Comment:
This should be in a `LaunchedEffect` not just in the @Composable because it'll be called on every recomposition.
See how its done for the theme.

```
LaunchedEffect(intent) {
  // fire the event here
}
```

Reply:
my bad ,In fact the recomposition is the reason navigation was invoked again , so checking for intent action after composable doesn't ensure the flow is subscribed ..
so i can only think of two solutions to invoke this navigation after the flow is subscribed ,using replay cache or a callback from the navigation root when the subscription count changes to 1.
If there is a way to do this without these complexities ,i would be happy to rework it that way.

Reply:
```
LaunchedEffect(intent) {
  // fire the event here
}
```

Should work just fine because this side-effect will be called only when the `intent` changes. Did you try it?

Reply:
yes i did try it, please correct me if i am wrong ,,
user presses shortcut -> on create called -> launched effect called -> navigation event invoked ->navigation root subscribes and starts listening for events ..so navigation event is lost.
so firing the event inside of side effect when intent changes doesn't do anything except for ensuring its not being called for every recomposition .

Reply:
@shashank-0-0 Hmm, that's weird. I don't have time but I'll investigate later. Try to find a way to put it in such way so it works. If not, I'll help you out.